src/notify/index.ts

```diff
@@ -113,29 +132,117 @@ export async function handler () {
         continue
       }
 
-      const videoIds = []
-      const needGetStartTimeVideos: Set<string> = new Set()
+      videoIdsPerChannels[channelId] = []
+      needGetStartTimeVideos[channelId] = new Set()
       notifyVideoData[channelId] = { title: feed.title, videos: {} }
 
       for (const item of feed.items) {
         const videoId = item.id.replace(/^yt:video:/, '')
         notifyVideoData[channelId].videos[videoId] = {
-          title: item.title,
+          videoTitle: item.title,
           updatedTime: new Date(item.updated),
           needInsert: true,
           isUpdated: false,
           isLiveStreaming: true,
-          privacyStatus: PrivacyStatus.Public
+          privacyStatus: PrivacyStatus.Public,
+          isCollab: false
+        }
+        videoIds.add(videoId)
+        videoIdsPerChannels[channelId].push(videoId)
+        needGetStartTimeVideos[channelId].add(videoId)
+      }
+    }
+
+    for (const { channel_id: { S: channelId }, twitter_id: { S: twitterId } } of channels) {
+      if (channelId === undefined) {
+        continue
+      }
+
+      if (twitterId !== undefined) {
+        await sleep(1000)
+        console.log('get twitter user timeline:', twitterId)
+        const timeLine = await twitterApiReadOnly.v2.userTimeline(twitterId, { max_results: 30 })
+        const tweetStack = timeLine.tweets
+        const tweets: Array<{ url: string, createdAt: string | undefined }> = []
+
+        while (tweetStack.length > 0) {
+          const tweet = tweetStack.pop()
+
+          if (tweet === undefined) {
+            continue
+          }
+
+          for (const shortUrl of tweet.text.matchAll(/https:\/\/t\.co\/[a-zA-Z0-9]+/g)) {
+            await sleep(1000)
+            const tweetUrl = shortUrl[0]
+            let url
+
+            try {
+              console.log('get:', tweetUrl)
+              const response = await axios.get(tweetUrl)
+              url = response.request.res.responseUrl
+            } catch (e) {
+              console.log(e)
+              continue
+            }
+            const twitterIdPattern = url.match(/https:\/\/twitter\.com\/[^/]+\/status\/([^/]+)/)
```

Comment:
```suggestion
            const twitterIdPattern = url.match(/^https:\/\/twitter\.com\/[^/]+\/status\/([^/]+)$/)
```
なんか攻撃に使われそうだな、この辺。
https://qiita.com/tamura__246/items/f41d2f69f1c8494493e0

Reply:
^, $はまずそうだけど、例えば `http://example.com/attack?url=https://twitter.com/Goryudyuma/status/` みたいなURLを踏ませられないかなーって。

Reply:
いやごめん、これはセーフな例だ。

Reply:
https://twitter.com/Goryudyuma/status/[攻撃コード]
みたいなことできないかなーと思ったけど、結局これ `twitterApiReadOnly.v2.tweets(tweetId)` に突っ込んでるだけだし大丈夫そう。